secadm/secadm.c

```diff
@@ -125,6 +151,111 @@ check_bsd(void)
 }
 
 static int
+featuresact(int argc, char *argv[])
+{
+	char *features[5] = { "aslr", "mprotect", "pageexec", "segvguard", NULL };
+	size_t sz = sizeof(int);
+	int value, i = 0, j;
+	char name[40];
+
+	printf("[+] available features:");
+	if (feature_present(FEATURE_PAX_ASLR))
+		printf(" ASLR");
+	if (feature_present(FEATURE_PAX_MPROTECT))
+		printf(" MPROTECT");
+	if (feature_present(FEATURE_PAX_PAGEEXEC))
+		printf(" PAGEEXEC");
+	if (feature_present(FEATURE_PAX_SEGVGUARD))
+		printf(" SEGVGUARD");
+
+	putchar('\n');
+
+	printf("[+] enabled features:");
+	do {
+		sprintf(name, "hardening.pax.%s.status", features[i]);
```

Comment:
I know this isn't user-controlled, but please use snprintf. I always do double-takes whenever I see sprintf().